sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +748,146 @@ def update_repository_properties(self, *args, **kwargs):
                 repository, value=properties._to_generated(), **kwargs  # pylint: disable=protected-access
             )
         )
+
+    @distributed_trace
+    def upload_manifest(
+        self, repository: str, manifest: "Union['OCIManifest', 'IO']", *, tag: "Optional[str]" = None, **kwargs: "Any"
+    ) -> None:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :returns: None
+        :rtype: None
+        :raises ValueError: If the parameter repository or manifest is None.
+        """
+        if (repository or manifest) is None:
+            raise ValueError("The parameter repository and manifest cannot be None.")
```

Comment:
Is there a reason we're doing client-side validation here?
Normally we would just let these values go to the service and let the service raise an error? Though sometimes if that error is user-unfriendly, we might do this...

Reply:
Updated.